9.2/run-postgresql.sh

```diff
@@ -1,12 +1,23 @@
 #!/bin/bash
 
+# Set HOME variable
+export HOME=/var/lib/pgsql
+
 # For SCL enablement
 source $HOME/.bashrc
 
 set -eu
 
 # Data dir
-export PGDATA=$HOME/data
+if [ -O $HOME/data ]
+then
+  export PGDATA=$HOME/data
+else
+  # If current user does not own data directory
+  # create a subdirectory that the user does own
+  mkdir $HOME/data/userdata
+  export PGDATA=$HOME/data/userdata
```

Comment:
In the case where $HOME/data is on an NFS mount, ideally the uid that's running the container owns the directory. If at worst they only have write permission, then the same logic applies... a subdirectory will be created that's owned by the current uid.